platform.md

```diff
@@ -170,30 +611,103 @@ A more thorough explanation is provided in the [Buildpack Interface Specificatio
+
+| Env Variable      | Layer Path   | Contents
+|-------------------|--------------|------------------
+| `PATH`            | `/bin`       | binaries
+| `LD_LIBRARY_PATH` | `/lib`       | shared libraries
+| `LIBRARY_PATH`    | `/lib`       | static libraries
+| `CPATH`           | `/include`   | header files
+| `PKG_CONFIG_PATH` | `/pkgconfig` | pc files
 
-User-provided environment variables intended for build and launch SHOULD NOT come from the same list.
-The end-user SHOULD be encouraged to define them separately.
-The platform MAY determine the initial environment of the build phase, detection phase, and launch.
-The lifecycle MUST NOT assume that all platforms provide an identical environment.
+The platform SHOULD NOT assume any other stack-provided environment variables are inherited by the buildpack.
 
-### Run Image Rebasing
+##### User-Provided Variables
+User-provided environment variables MUST be supplied by the platform as files in the `<platform>/env/` directory.
+Each file SHALL define a single environment variable, where the file name defines the key and the file contents define the value.
 
-Run image rebasing allows for fast stack updates for already-exported OCI images with minimal data transfer when those images are stored on a Docker registry.
-When a new stack version with the same stack ID is available, the app layers and launch layers SHOULD be rebased on the new run image by updating the image's configuration to point at the new run image.
-Once the new run image is present on the registry, filesystem layers SHOULD NOT be uploaded or downloaded.
+User-provided environment variables MAY be modified by prior buildpacks before they are provided to a given buildpack.
 
-The new run image MUST have an identical stack ID and MUST include the exact same set of mixins.
+The platform SHOULD NOT set user-provided environment variables directly in the lifecycle execution environment.
 
-![Launch](img/launch.svg)
+#### Launch Environment
+User-provided modifications to the process execution environment SHOULD be set directly in the lifecycle execution environment.
+
+The process SHALL inherit both stack-provided and user-provided variables from the lifecycle execution environment with the following exceptions:
+* `CNB_APP_DIR`, `CNB_LAYERS_DIR` and `CNB_PROCESS_TYPE` SHALL NOT be set in the process execution environment.
+* The lifecycle SHALL apply buildpack-provided modifications to the environment as outlined in the [Buildpack Interface Specification](buildpack.md).
 
 ### Caching
 
-Each platform SHOULD implement caching so as to appropriately optimize performance.
+IF caching is enabled the platform is responsible for providing the lifecycle with access to the correct cache.
+Whenever possible, the platform SHOULD provide the same cache to each rebuild of a given app image.
 Cache locality and availability MAY vary between platforms.
 
 ## Data Format
 
-### order.toml (TOML)
+### Files
```

Comment:
Maybe it's worth adding an example of each file (or pointing to a location in the samples repository with such examples).

Reply:
I think we should provide examples outside the spec in more friendly formats like (future) lifecycle docs.

Reply:
Sounds good but don't you think it's worth adding a link in the spec to these examples?
Anyway, if these examples don't exist yet, we can discuss it when it'll be relevant :)